src/main/java/gregicality/science/api/capability/IPressureContainer.java

```diff
@@ -0,0 +1,195 @@
+     * @return true if the change is safe, else false
+     */
+    default boolean changeParticles(double amount, boolean simulate) {
+        if (simulate) return isPressureSafe(getPressureForParticles(getParticles() + amount));
+        setParticles(getParticles() + amount);
+        return isPressureSafe();
+    }
+
+    /**
+     * @return the minimum pressure this container can handle
+     */
+    double getMinPressure();
+
+    /**
+     * @return the maximum pressure this container can handle
+     */
+    double getMaxPressure();
+
+    /**
+     * @return true if the pressure is safe for the container, else false
+     */
+    default boolean isPressureSafe() {
+        return isPressureSafe(getPressure());
+    }
+
+    /**
+     * @param pressure the pressure to check
+     * @return true if the pressure is safe for the container, else false
+     */
+    default boolean isPressureSafe(double pressure) {
+        return pressure < getMaxPressure() && pressure > getMinPressure();
+    }
+
+    /**
+     * @return if the pressure is around atmospheric levels
+     */
+    default boolean isNormalPressure() {
+        return Math.abs(getPressure() - GCYSValues.EARTH_ATMOSPHERIC_PRESSURE) < PRESSURE_TOLERANCE;
+    }
+
+    /**
+     *
+     * @return if the pressure is a vacuum
+     */
+    default boolean isVacuum() {
+        return getPressure() < GCYSValues.EARTH_ATMOSPHERIC_PRESSURE - PRESSURE_TOLERANCE;
+    }
+
+    /**
+     * Causes an explosion due to pressure
+     *
+     * @param world the world to create the explosion in
+     * @param pos   the position of the explosion
+     */
+    default void causePressureExplosion(World world, BlockPos pos) {
+        if (world != null && !world.isRemote) {
+            final float explosionPower = (float) Math.abs(Math.log(getPressure()));
+            world.setBlockToAir(pos);
+
+            if (isVacuum()) {
+                world.createExplosion(null, pos.getX() + 0.5, pos.getY() + 0.5, pos.getZ() + 0.5,
+                        explosionPower, false);
+            } else {
+                world.createExplosion(null, pos.getX() + 0.5, pos.getY() + 0.5, pos.getZ() + 0.5,
+                        explosionPower, ConfigHolder.machines.doesExplosionDamagesTerrain);
+            }
+        }
+    }
+
+    /**
+     * Equalizes the pressure between containers. This does not modify volume.
+     *
+     * @param containers the containers to merge
+     */
+    static void mergeContainers(@Nonnull IPressureContainer... containers) {
+        mergeContainers(true, containers);
+    }
+
+    /**
+     * Equalizes the pressure between containers. This does not modify volume.
+     *
+     * @param checkSafety whether to check if changing pressure is safe before modifying container values
+     * @param containers  the containers to merge
+     */
+    static void mergeContainers(boolean checkSafety, @Nonnull IPressureContainer... containers) {
```

Comment:
This is probably often called with a source and target container. However if there are multiple pressure generators and consumers on a net, each will try to equalize their container with the net on each insertion/extraction this will cause the pressure to fluctuate a lot.